src/plots/cartesian/layout_defaults.js

```diff
@@ -164,13 +169,7 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
 
         handlePositionDefaults(axLayoutIn, axLayoutOut, coerce, positioningOptions);
 
-        layoutOut[axName] = axLayoutOut;
-
-        // so we don't have to repeat autotype unnecessarily,
-        // copy an autotype back to layoutIn
-        if(!layoutIn[axName] && axLayoutIn.type !== '-') {
-            layoutIn[axName] = {type: axLayoutIn.type};
-        }
+        axLayoutOut._input = axLayoutIn;
```

Comment:
This is essentially the new `ax_gd`. This pattern is similar to other components that require a ref to the user container to mutate it on interaction (e.g. updatemenus and sliders). Moreover, IMHO this pattern is a lot better than having to keep the full `gd` in-sync.